packages/devtools_app/lib/src/screens/memory/shared/heap/class_filter.dart

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import '../../primitives/class_name.dart';
+
+enum ClassFilterType {
+  all,
+  except,
+  only,
+}
+
+/// What should be done to apply new filter to a set of data.
+enum FilteringTask {
+  /// New filter is equivalent to previous, so nothing should be done.
+  doNothing,
+
+  /// Previous filtering results cannot be reused.
+  refilter,
+
+  /// Previous filtering results can be reused.
+  reuse,
+}
+
+@immutable
+class ClassFilter {
+  ClassFilter({
+    required this.filterType,
+    required String except,
+    required String? only,
+  })  : except = _trimByLine(except),
+        only = only == null ? null : _trimByLine(only);
+
+  ClassFilter.empty()
+      : this(
+          filterType: ClassFilterType.all,
+          except: '$coreLibrariesAlias\n$dartAndFlutterLibrariesAlias',
+          only: null,
+        );
+
+  static String _trimByLine(String value) =>
+      value.split('\n').map((e) => e.trim()).join('\n');
+
+  static const String dartAndFlutterLibrariesAlias = '\$dart-flutter-libraries';
+  static const String coreLibrariesAlias = '\$core-libraries';
+
+  final ClassFilterType filterType;
+  final String except;
+
+  /// If the value is null, it should be initialized before displaying.
+  final String? only;
+
+  bool get isEmpty =>
+      filterType == ClassFilterType.all ||
+      (filterType == ClassFilterType.except && except.trim().isEmpty) ||
+      (filterType == ClassFilterType.only && (only ?? '').trim().isEmpty);
+
+  String get buttonTooltip {
+    if (isEmpty) return 'Filter classes and packages.';
+    return displayString;
+  }
+
+  bool equals(ClassFilter value) {
+    return value.filterType == filterType &&
+        value.except == except &&
+        value.only == only;
+  }
+
+  Set<String> _filtersAsSet() {
```

Comment:
one thing to be aware of here is that users may expect to be able to enter a comma separated list instead of adding a new line for each filter item

Reply:
added support for comma